pkg/dockerfile/dockerfile.go

```diff
@@ -58,6 +58,10 @@ func ParseStages(opts *config.KanikoOptions) ([]instructions.Stage, []instructio
 		return nil, nil, errors.Wrap(err, "parsing dockerfile")
 	}
 
+	if opts.CacheCopyLayers && len(stages) >= 2 {
```

Comment:
Is there anything we can do besides just fail here? I don't know how many builds out there depend on this, and while I agree failing is better than silently doing the wrong thing, both are worse than doing the right thing, if we can do that.

Reply:
@imjasonh short of properly supporting caching copy layers in multistage builds, no, there's else nothing we can do here. I personally don't know enough to add this support. I believe it would be fairly complicated.

Reply:
Actually, I guess maybe we could quietly disable CacheCopyLayers here, if that would work. We could print a message about this, but I think that for most user setups, this message wouldn't be seen. I'm dubious about this, but I guess it's an option.

The issue is that I don't know the kaniko code well. So I'm not 100% sure if any parts of the codebase have already read the value of CacheCopyLayers (or whether parts might be added which would have already read it). So this seems like a way to maybe silently reintroduce the errors.

Reply:
That could work. A slower correct build is also better IMO than a fast silently-broken one, or a fails-immediately one.

Reply:
Do you have a test build that demonstrates the broken behavior? You could run it through kaniko with your change to disable `CacheCopyLayers` and see if it comes out as expected or not.

If we can't figure out how to fix it, we can just fail fast like you have before.